diff --git a/src/mesa/main/draw.c b/src/mesa/main/draw.c
index 15cef9e..fb5e184 100644
--- a/src/mesa/main/draw.c
+++ b/src/mesa/main/draw.c
@@ -448,7 +448,7 @@ static GLenum
 validate_draw_arrays(struct gl_context *ctx,
                      GLenum mode, GLsizei count, GLsizei numInstances)
 {
-   if (count < 0 || numInstances < 0)
+   if (count <= 0 || numInstances < 0)
       return GL_INVALID_VALUE;
 
    GLenum error = _mesa_valid_prim_mode(ctx, mode);
@@ -497,6 +497,9 @@ _mesa_validate_DrawArraysInstanced(struct gl_context *ctx, GLenum mode, GLint fi
       error = validate_draw_arrays(ctx, mode, count, numInstances);
    }
 
+   if (!numInstances)
+	   error = GL_INVALID_VALUE;
+
    if (error)
       _mesa_error(ctx, error, "glDrawArraysInstanced");
 
@@ -563,7 +566,7 @@ _mesa_validate_DrawElementsInstanced(struct gl_context *ctx,
                                      GLsizei numInstances)
 {
    GLenum error =
-      validate_DrawElements_common(ctx, mode, count, numInstances, type);
+      validate_DrawElements_common(ctx, mode, count, numInstances, type) && (numInstances > 0);
 
    if (error)
       _mesa_error(ctx, error, "glDrawElementsInstanced");
@@ -597,6 +600,9 @@ _mesa_validate_DrawTransformFeedback(struct gl_context *ctx,
       }
    }
 
+   if (!numInstances)
+	   error = GL_INVALID_VALUE;
+
    if (error)
       _mesa_error(ctx, error, "glDrawTransformFeedback*");
 
